### PR TITLE
Improve PIN input component

### PR DESCRIPTION
**Pull request by nmattia**
This improves the component in two ways:

* The component does not have `type="password"` anymore but instead may use a custom font that replaces all chars with the dot symbol. This ensures the (currently 6) inputs do not independently trigger the browser's password manager (which would only save a single char)
* The component may autofocus the first input

<hr/><details><summary>🟡 Some screens were changed</summary></details>

### CODE REVIEW

File: src/frontend/src/components/pinInput.ts [hunk 1]
```diff
@@ -112,15 +116,21 @@ export const pinInput = <T>({
...
         ${Array.from(
           { length: pinLength },
+          /* XXX: we use a special class/font for the 'secret' PIN input instead of setting
+           * type="password", otherwise the browser tries to save one char as a password */
           (_, _ix) => html`
             <li class="c-list--pin-char c-input--anchor">
               <label class="c-input--anchor__wrap">
                 <input
+                  ?autofocus=${focus}
                   autocomplete="off"
```
Comment (peterpeterparker) [inline src/frontend/src/components/pinInput.ts:126]: should `autocapitalize` and `autocorrect` be turned `off` is the input becomes `text`?

File: src/frontend/src/styles/main.css [hunk 2]
```diff
@@ -69,6 +69,15 @@
   font-display: swap;
 }
 
+/* Workaround font for pretending a field is a password without setting type="password"
+ * (all chars in the font are the dot symbol) */
+@font-face {
+  font-family: "password";
+  font-style: normal;
+  font-weight: 400;
+  src: url("../../assets/fonts/password.woff2") format("woff2");
```
Comment (peterpeterparker) [inline src/frontend/src/styles/main.css:78]: license wise can we use this font?

File: src/frontend/src/styles/main.css [hunk 3]
```diff
@@ -1763,6 +1772,11 @@ by all browsers (FF is missing) */
   border-color: red;
 }
 
+/* Use a fake font for "password"-style pin input */
+.c-input--pin__hidden {
```
Comment (peterpeterparker) [inline src/frontend/src/styles/main.css:1776]: not familiar with the style guide, but wouldn't a a class name `pin__password` be more meaningful that `hidden`? spontaneously I would assume `display: hidden` is used but again, do not know the css here.

File: src/showcase/src/components.ts [hunk 4]
```diff
@@ -50,21 +50,39 @@ const mkToast = (): TemplateResult => {
 
 const choosePin = (): TemplateResult => {
   const submittedPin = new Chan("N/A");
+  const pinInputHidden_ = pinInput({
+    verify: (pin: string) => ({ ok: true, value: pin }),
+    onSubmit: (pin) => submittedPin.send(pin),
```
Comment (peterpeterparker) [inline src/showcase/src/components.ts:55]: `(ping: string)` or type is inteferred?

File: src/frontend/src/styles/main.css (hunk 2 above)
Comment (nmattia) [reply src/frontend/src/styles/main.css:78]: working with a designer to get one, good catch

File: src/frontend/src/components/pinInput.ts [hunk 5]
```diff
@@ -6,13 +6,15 @@ import { TemplateResult, html } from "lit-html";
 import { asyncReplace } from "lit-html/directives/async-replace.js";
 import { Ref, createRef, ref } from "lit-html/directives/ref.js";
 
-type Result<T> = { ok: true; value: T } | { ok: false; error: string };
+export type Result<T> = { ok: true; value: T } | { ok: false; error: string };
```
[the remark this answers is not in the document]
Comment (nmattia) [reply src/frontend/src/components/pinInput.ts:9]: Un-exported; will export when I actually use it elsewhere

File: src/frontend/src/components/pinInput.ts (hunk 1 above)
Comment (nmattia) [reply src/frontend/src/components/pinInput.ts:126]: good catch (modulo `autocorrect`/`spellcheck`)

File: src/frontend/src/styles/main.css (hunk 3 above)
Comment (nmattia) [reply src/frontend/src/styles/main.css:1776]: good catch, called it `__secret`

**Review by peterpeterparker — APPROVED** [review]
LGTM minus the font which will be replaced before merge. Thanks.

File: src/frontend/src/styles/main.css (hunk 2 above)
Comment (nmattia) [reply src/frontend/src/styles/main.css]: made a custom font, thanks!

File: src/internet_identity/src/assets.rs [hunk 6]
```diff
@@ -259,6 +259,7 @@ fn collect_assets_from_dir(dir: &Dir) -> Vec<(String, Vec<u8>, ContentEncoding,
             "png" => (file_bytes, ContentEncoding::Identity, ContentType::PNG),
             "svg" => (file_bytes, ContentEncoding::Identity, ContentType::SVG),
             "webp" => (file_bytes, ContentEncoding::Identity, ContentType::WEBP),
+            "woff2" => (file_bytes, ContentEncoding::GZip, ContentType::WOFF2),
```
Comment (frederikrothenberger) [inline src/internet_identity/src/assets.rs:262]: Is `ContentEncoding::GZip` accurate for non-gzipped `woff2`?

Comment (nmattia) [reply src/internet_identity/src/assets.rs:262]: No it's not! copy pasta

Comment (nmattia) [reply src/internet_identity/src/assets.rs:262]: https://github.com/dfinity/internet-identity/pull/1859

**Pull request merged**